src/chunk.c

```diff
@@ -133,6 +133,7 @@ static void init_scan_by_qualified_table_name(ScanIterator *iterator, const char
 #define ASSERT_IS_VALID_CHUNK(chunk) Assert(IS_VALID_CHUNK(chunk))
 
 #define ASSERT_IS_NULL_OR_VALID_CHUNK(chunk) Assert(chunk == NULL || IS_VALID_CHUNK(chunk))
+static void lock_referenced_tables(Oid table_relid);
```

Comment:
It is trivial to ignore this line when reviewing and go to the actual function. I will remove it if I have to do other updates, otherwise I'll leave it.

Reply:
Moving the implementation before function, which uses it, will simplifies reviewing, while ignoring this line doesn't help.

If you hesitant to move the implementation, then I insist to move this declaration line to the same area as other declarations. I don't see why it is good to place it separately and not together with other declarations.
This is common practice to group declarations together. See already existing group at https://github.com/timescale/timescaledb/blob/master/src/chunk.c#L108-L117

Reply:
Removed.